chpl/chpl-service/src/main/java/gov/healthit/chpl/validation/certifiedProduct/CertifiedProduct2014Validator.java

```diff
@@ -390,39 +396,42 @@ public void validate(CertifiedProductSearchDetails product) {
         if (StringUtils.isEmpty(product.getReportFileLocation())) {
             product.getErrorMessages().add("Test Report URL is required but was not found.");
         }
+        if(product.getCertificationStatus().isEmpty()){
+        	product.getErrorMessages().add(getErrorMessage("listing.missingCertificationStatus"));
+        }
+        if(StringUtils.isEmpty(product.getTestingLab())){
+        	product.getErrorMessages().add(getErrorMessage("listing.missingTestingLab"));
+        }
+        if(product.getHasQms() && product.getQmsStandards().isEmpty()){
```

Comment:
The "has QMS" boolean as used by the pending objects (for new uploads) exists because the upload file has two columns for QMS. It's weird to have two columns but they do anyway - one that has a boolean value whether or not the listing has QMS and the other that has multiple rows of  specific QMS standards written out. I think you correctly added the check for the new uploads (line 222) but I don't think the same check should exist in this section (where CertifiedProductSearchDetails is being validated) because this code runs when editing a listing rather than on a new upload. An uploaded and confirmed listing that is being edited does not have that boolean QMS value - it just either has a list of QMS standards or it doesn't. I think that your addition at line 222 is good as well as the changes to PendingCertifiedProductDTO but the rest of the places where stuff like "hasQms" was added code are probably not necessary.